thingsboard_gateway/connectors/ble/device.py

```diff
@@ -0,0 +1,249 @@
+#     limitations under the License.
+
+from threading import Thread
+from platform import system
+from time import time, sleep
+import asyncio
+
+from bleak import BleakClient
+
+from thingsboard_gateway.connectors.connector import log
+
+MAC_ADDRESS_FORMAT = {
+    'Darwin': '-',
+    'other': ':'
+}
+
+
+class Device(Thread):
+    def __init__(self, config):
+        super().__init__()
+        self.loop = None
+        self.stopped = False
+        self.name = config['name']
+        self.device_type = config.get('deviceType', 'default')
+        self.timeout = config.get('timeout', 10000) / 1000
+        self.show_map = config.get('showMap', False)
+
+        self.daemon = True
+
+        try:
+            self.mac_address = self.validate_mac_address(config['MACAddress'])
+            self.client = BleakClient(self.mac_address)
+        except ValueError as e:
+            self.client = None
+            self.stopped = True
+            log.error(e)
+
+        self.poll_period = config.get('pollPeriod', 5000) / 1000
+        self.config = {
+            'telemetry': config.get('telemetry', []),
+            'attributes': config.get('attributes', []),
+            'attributeUpdates': config.get('attributeUpdates', []),
+            'serverSideRpc': config.get('serverSideRpc', [])
+        }
+        self.callback = config['callback']
+        self.last_polled_time = 0
+
+        self.notifying_chars = []
+
+        self.start()
+
+    @staticmethod
+    def validate_mac_address(mac_address):
+        os_name = system()
+
+        if MAC_ADDRESS_FORMAT[os_name if os_name == 'Darwin' else 'other'] not in mac_address:
+            raise ValueError(f'Mac-address is invalid for {os_name} os')
+
+        return mac_address.upper()
+
+    async def timer(self):
+        await self.__process_self()
+        self.last_polled_time = time()
+
+        while True:
+            if time() - self.last_polled_time >= self.poll_period:
+                self.last_polled_time = time()
+                await self.__process_self()
+
+            await asyncio.sleep(.2)
+
+    async def notify_callback(self, sender: int, data: bytearray):
+        not_converted_data = {'telemetry': [], 'attributes': []}
+        for section in ('telemetry', 'attributes'):
+            for item in self.config[section]:
+                if item.get('handle') and item['handle'] == sender:
+                    not_converted_data[section].append({'data': data, **item})
+
+                    data_for_converter = {
+                        'deviceName': self.name,
+                        'deviceType': self.device_type,
+                        'config': {
+                            'attributes': self.config['attributes'],
+                            'telemetry': self.config['telemetry']
+                        },
+                        'data': not_converted_data
+                    }
+
+                    self.callback(data_for_converter)
+
+                    return
```

Comment:
What if we want to read one characteristic as a telemetry and an attribute keys?